### ISSUE TITLE
Active Link Class not added when using append option

### ISSUE THREAD

**Issue by Awea** [vuejs/vue-router#498]
### Vue.js & vue-router.js version
1.0.24, 0.7.13

I expect that the link is active when I navigate to a route, the `v-link` component is defined with the option `append:true`.

I've made a small [Codepend](https://codepen.io/Awea/pen/grePEj) to make it clearer. It's seems to come from the regex used to test the route and add the class.

**Comment by LinusBorg** [comment]
I don't think a v-link with  `append: true` can ever have the `active` state, logically.

* To stay with your example, we start at `/` and go to `/foo`
* now the second link, with `append: true` points to `/foo/bar`
* we click it and navigate to `/foo/bar`
* but now the same link points to `/foo/bar/bar` - because it is *always relative to the current URL*

So, applying v-link-active in this situation would be a confusing situation, because the user would by convention expect that this link points to `/foo/bar`, which is the currently active URL - but it doesn't, it points to `/foo/bar/bar`

Long story short: links with `append: true` are not meant to be used to be used in things like navbars, where they stay during page changes and indicate the active location. They should rather be used to re-use components that link to re-usable subComponents, like making a button component that navigates to a `comments` subcomonent and can be re-used as a drop-in in many different components that want to show comments without having to adjust paths. `/post/1/comments`, `photo/1/comments

**Comment by fnlctrl** [comment]
Closing due to inactivity.
